temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivitySlotSupplierQueue.java

```diff
@@ -0,0 +1,117 @@
+package io.temporal.internal.worker;
+
+import io.temporal.worker.tuning.LocalActivitySlotInfo;
+import io.temporal.worker.tuning.SlotPermit;
+import io.temporal.workflow.Functions;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class LocalActivitySlotSupplierQueue {
+  static final class QueuedLARequest {
+    final boolean isRetry;
+    final SlotReservationData data;
+    final LocalActivityAttemptTask task;
+
+    QueuedLARequest(boolean isRetry, SlotReservationData data, LocalActivityAttemptTask task) {
+      this.isRetry = isRetry;
+      this.data = data;
+      this.task = task;
+    }
+  }
+
+  private final PriorityBlockingQueue<QueuedLARequest> requestQueue;
+  private final Semaphore newExecutionsBackpressureSemaphore;
+  private final TrackingSlotSupplier<LocalActivitySlotInfo> slotSupplier;
+  private final Functions.Proc1<LocalActivityAttemptTask> afterReservedCallback;
+  private final Thread queueThread;
+  private static final Logger log =
+      LoggerFactory.getLogger(LocalActivitySlotSupplierQueue.class.getName());
+  private volatile boolean running = true;
+
+  LocalActivitySlotSupplierQueue(
+      TrackingSlotSupplier<LocalActivitySlotInfo> slotSupplier,
+      Functions.Proc1<LocalActivityAttemptTask> afterReservedCallback) {
+    this.afterReservedCallback = afterReservedCallback;
+    // TODO: See if I can adjust this for dynamic ones based on current rate
```

Comment:
This is kinda the last thing. I don't think there's really much sensible that can be done here, but maybe we want to make it an option?

Reply:
Yeah  I would open an issue to revisit the backoff logic. My  main goal was to  just maintain the current behavior for a fixed slot suppliers without some more thought.